### ISSUE TITLE
`escaped` hangs if `normal` parser accepts empty

### ISSUE THREAD

**Issue by FauxFaux** [Geal/nom#1336]
If one uses `alpha0` as a "normal" (first argument) parser for `escaped`, the parser hangs indefinitely if given a non-matching input.

e.g. `escaped(alpha0, '\\', one_of("n"))("7")` hangs.

The loop is here; `i2` == `i` at all times, as the `normal` parser returns the whole input.

https://github.com/Geal/nom/blob/e9ba351567d0882a771f01436eda6ed27dca16a1/src/bytes/complete.rs#L497-L501

(This could be a documentation bug.)

## Prerequisites

Here are a few things you should provide to help me understand the issue:

- Rust version : `rustc 1.54.0-nightly (ed597e7e1 2021-06-08)`
- nom version : 6.2.1
- nom compilation features used: default

## Test case

```
fn escaped_string(input: &str) -> IResult<&str, &str> {
    use nom::bytes::complete::escaped;
    use nom::character::complete::{alpha0, one_of};
    escaped(alpha0, '\\', one_of("n"))(input)
}

#[test]
fn escaped_hang() { escaped_string("7").unwrap(); }
```

**Comment by Bytekeeper** [comment]
Actually, on the same nom version, it hangs for me even with valid input:
```rust
escaped_string("a7").unwrap()
```
should result in `Ok(("7", "a"))` - but will also hang.

**Comment by Geal** [comment]
this should be fixed by d31a752 in nom 7.0

**Comment by Geal** [comment]
fix released in nom 7: https://crates.io/crates/nom/7.0.0
